### DISCUSSION TITLE
Three level tesselation

### DISCUSSION THREAD

**Discussion by danishTUE** [neperfepx/neper#901] [Q&A] [answered]
Hello Romain,

I am interested to generate a three level tesselation: let say at level _l1_, _l2_ and _l3_. _l1_ and _l2_ are equi-axed while _l3_ is supposed to be of lamellar morphology, dividing every cell at level _l2_ into equi-width _m_ number of lamellae along a particular direction _d_.  _m_ is fixed and does not vary with cells at _l2_  however _d_ changes from cell to cell at level _l2_. I understand that in order to invoke lamellae morphology, we need to use _from_morpho_  while defining _n_ (number of cells). However, this way I can not fix the number of lamellae to be _m_. One possibility that I can think of is to first run a two level tesselation generation and then generate the statistics for the resultant cells using -statcell to get the idea of the cell size. Then re-run the three level tesselation, with lamellae width now defined using a file (based on the information obtained from -statcell). I have four queries:

1) Is there a more straight forward way to achieve the desired outcome that I described above then running Neper two times?
2) When I use -statcell as _-statcell 'id','size','diameq'_ I get a file with information based on the id of the cell. However I do not have the the information of the hierarchy of the tesselation (i.e. I do not know which cell of _l3_ belongs to which cell at level _l2_). How can I get this information?
3) How can I get the dimension of a cell along a particular direction _d_ (This information will help me in dividing the cell at level _l2_ into equi-width _m_ lamellae along direction _d_) 
4) Can you please share a simple quick script for your suggestion?

Regards,

Danish

**Comment by rquey** [comment, marked as answer]
In the last version (`4.9.1-4`), I have "unlocked" `-n ... -morpho lamellar`, which I think solves your problem altogether. Please get back to me if it doesn't.
```
neper -T -n 10 -morpho lamellar -dim 2 -o tmp
neper -V tmp.tess -print img
```
![img](https://github.com/user-attachments/assets/340d406b-58e2-4a55-be99-f9dd94ad961c)

```
neper -T -n 10::10 -morpho gg::lamellar -dim 2 -o tmp
neper -V tmp.tess -print img
```
![img](https://github.com/user-attachments/assets/cf57c6e5-032c-4a25-b83c-8e29067d636c)

**Reply by danishTUE** [reply]
Many thanks! I will try this feature and let you know if I have some issues.
